test/PoolManager.t.sol

```diff
@@ -948,42 +948,40 @@ contract PoolManagerTest is Test, Deployers, GasSnapshot {
     function test_setProtocolFee_updatesProtocolFeeForInitializedPool(uint24 protocolFee) public {
         (,, uint24 slot0ProtocolFee,) = manager.getSlot0(key.toId());
         assertEq(slot0ProtocolFee, 0);
-        feeController.setProtocolFeeForPool(key.toId(), protocolFee);
 
         uint16 fee0 = protocolFee.getZeroForOneFee();
         uint16 fee1 = protocolFee.getOneForZeroFee();
+        vm.prank(address(feeController));
         if ((fee0 > 2500) || (fee1 > 2500)) {
-            vm.expectRevert(IProtocolFees.ProtocolFeeControllerCallFailedOrInvalidResult.selector);
-            manager.setProtocolFee(key);
+            vm.expectRevert(IProtocolFees.InvalidProtocolFee.selector);
+            manager.setProtocolFee(key, protocolFee);
         } else {
             vm.expectEmit(false, false, false, true);
             emit IProtocolFees.ProtocolFeeUpdated(key.toId(), protocolFee);
-            manager.setProtocolFee(key);
+            manager.setProtocolFee(key, protocolFee);
 
             (,, slot0ProtocolFee,) = manager.getSlot0(key.toId());
             assertEq(slot0ProtocolFee, protocolFee);
         }
     }
 
-    function test_setProtocolFee_failsWithInvalidProtocolFeeControllers() public {
+    function test_setProtocolFee_failsWithInvalidFee() public {
         (,, uint24 slot0ProtocolFee,) = manager.getSlot0(key.toId());
         assertEq(slot0ProtocolFee, 0);
 
-        manager.setProtocolFeeController(revertingFeeController);
-        vm.expectRevert(IProtocolFees.ProtocolFeeControllerCallFailedOrInvalidResult.selector);
-        manager.setProtocolFee(key);
-
         manager.setProtocolFeeController(outOfBoundsFeeController);
-        vm.expectRevert(IProtocolFees.ProtocolFeeControllerCallFailedOrInvalidResult.selector);
-        manager.setProtocolFee(key);
+        uint24 newProtocolFee = outOfBoundsFeeController.protocolFeeForPool(key);
+        vm.prank(address(outOfBoundsFeeController));
```

Comment:
i think i'd rather we add a function to the outOfBoundsFeeController contract that calls the PoolManager itself. And then here you'd do like
```
vm.expectRevert(IProtocolFees.InvalidProtocolFee.selector);
outOfBoundsFeeController.setProtocolFee(key);
```

and the contract would then call 
```
manager.setProtocolFee(key, newProtocolFee)
```

instead of pranking the contract address